src/compiler/bundle/ext-transforms-plugin.ts

```diff
@@ -131,11 +171,23 @@ export const extTransformsPlugin = (
           return s.styleTag === data.tag && s.styleMode === data.mode && s.styleText === cssTransformResults.styleText;
         });
 
-        if (!hasUpdatedStyle) {
+        /**
+         * if the style has updated, compose all styles for the component
+         */
+        if (!hasUpdatedStyle && cmpStyles && data.tag && data.mode) {
+          const externalStyles = cmp && cmp.styles[0] && cmp.styles[0].externalStyles;
+
+          /**
+           * if component has external styles, use a list to keep the order to which
+           * styles are applied
+           */
+          const styleText = externalStyles
+            ? externalStyles.map((es) => cmpStyles.get(es.absolutePath)).join('\n')
+            : [...cmpStyles.values()].join('\n');
```

Comment:
This is solely to allow us to keep a sense of order in these styles. A component like this:

```ts
@Component({
  tag: 'my-component',
  styleUrls: ['a.css', 'b.css'],
})
```
can look completely different as
```ts
@Component({
  tag: 'my-component',
  styleUrls: ['b.css', 'a.css'],
})
```

The map we build in `cmpStyles` is based on the order these styles are transformed but (I think) don't reflect the order they are defined. For that I found that `cmp?.styles?.[0].externalStyles` can be the right indicator. However the interface suggests that the property can be undefined. So I believe `[...cmpStyles.values()].join('\n')` will actually never be used but is needed here because of the way we define `ComponentCompilerMeta`.